msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/ResetPasswordStates.kt

```diff
@@ -311,7 +315,13 @@ class ResetPasswordPasswordRequiredState internal constructor(
                 return@withContext when (val result =
                     rawCommandResult.checkAndWrapCommandResultType<ResetPasswordSubmitNewPasswordCommandResult>()) {
                     is ResetPasswordCommandResult.Complete -> {
-                        ResetPasswordResult.Complete
+                        ResetPasswordResult.Complete(
+                            nextState = SignInAfterSignUpState(
```

Comment:
If we're going to reuse `SignInAfterSignUpState` state class (which I'd be in favour of), we need to rename it. The current name doesn't fit the SSPR flow.

Reply:
Good point - any naming suggestions? The most appropriate I can think of is `SignInAfterFlowCompletionState`.

Reply:
There was an idea a few months ago of not introducing a new state for this at all, but using the existing SignIn state (and adding new parameters, functions, etc.). Could that work?

Reply:
Not sure what that would look like - from what I know, we initiate the sign in flow by calling signIn from the public client application itself, then the sign in states we have inherit from BaseState and reflect the potential results from the initial sign in call, eg: `SignInCodeRequiredState`, `SignInPasswordRequiredState`, etc. So I don't see how we could extend or re-use those states for this case.

For now, it looks like there was an abstract class called `SIgnInAfterSignUpBaseState` that contains the logic for running the sign in call, so I've re-named that to `SignInAfterFlowCompletionState`, and created a `SignInAfterResetPasswordState` that extends that abstract class. Let me know what you think.

Reply:
What does the current setup (base class with 2 classes extending it) have that we couldn't do with just 1 class; the base class? I.e. why do we need `SignInAfterResetPasswordState` and `SignInAfterSignUpState`?

Reply:
Naming, mainly. Returning their respective names from the completion result (in my opinion) is more clear compared to the base name of `SignInAfterFlowCompletionState`.

Reply:
Can we rename the base class to make it useful for every flow? I'm questioning whether:
``` 
val result: ResetPassword.Complete
val nextState: SignInAfterResetPasswordState = result.nextState
nextState.signIn()
```

is much clearer then
```
val result: ResetPassword.Complete
val nextState: SignInAfterFlowCompletionState = result.nextState
nextState.signIn()
```

Developer should be checking the result class as against a list of expected results. The name of the `nextState` class is much less relevant (in many cases developers might not even see it). 
I do agree that `SignInAfterFlowCompletionState` is not a great name. Can we come up with something more developer friendly, something simpler? `QuickSignInState`?

Reply:
After discussing the naming, the new name will be `SignInContinuationState`. The `SignInAfterFlowCompletionState` is no longer abstract and is now used for both the Sign Up and Reset Password flows. `SignInAfterSignUpState` and `SignInAfterPasswordState` are both removed.